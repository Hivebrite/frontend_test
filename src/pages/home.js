import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

import List from '../components/list';
import Map from '../components/map';
import data from '../data/cities.json';


const Wrapper = styled.div`
    margin: 40px auto;
    padding: 30px;
    width: 1000px;
    height: 600px;
    border: 1px solid lightgrey;
    border-radius: 3px;
    display: flex;
    align-items: center;
`

const Home = () => {
    const [cities, setCities] = useState([]);
    const [filteredList, setFilteredList] = useState([]);
    const [selectedCity, setSelectedCity] = useState();

    const selectCity = (rank) => {
        const city = filteredList.find((e) => e.rank === rank);
        setSelectedCity(city);
    };

    const filterList = (str) => {
        console.log('filterList str:', str);
        if (str.length > 1) {
            const filter = cities.filter((e) => e.city.toLowerCase().indexOf(str.toLowerCase()) !== -1);
            setFilteredList(filter);
        } else {
            setFilteredList(cities.slice(0, 10));
        }
    }

    useEffect(() => {
        setCities(data);
    }, []);

    useEffect(() => {
        if (cities && cities.length > 0) {
            setFilteredList(cities.slice(0, 10));
        }
    }, [cities]);

    return (
        <Wrapper>
            <List cities={filteredList} selectCity={selectCity} searchList={filterList}/>
            <Map city={selectedCity}/>
        </Wrapper>
    );
};

export default Home;
