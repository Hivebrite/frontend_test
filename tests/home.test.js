import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Wrapper } from '../src/pages/home';
import List from '../src/components/list';
import Map from '../src/components/map';
import City from '../src/components/city';

configure({ adapter: new Adapter() });

describe('Home page', () => {
    it('should have 2 components', () => {
        const tree = renderer.create(
            <Wrapper>
                <List />
                <Map />
            </Wrapper>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    describe('List tests', () => {
        const cities = [
            {
                "city": "New York",
                "growth_from_2000_to_2013": "4.8%",
                "latitude": 40.7127837,
                "longitude": -74.0059413,
                "population": "8405837",
                "rank": "1",
                "state": "New York"
            },
            {
                "city": "Los Angeles",
                "growth_from_2000_to_2013": "4.8%",
                "latitude": 34.0522342,
                "longitude": -118.2436849,
                "population": "3884307",
                "rank": "2",
                "state": "California"
            },
            {
                "city": "Chicago",
                "growth_from_2000_to_2013": "-6.1%",
                "latitude": 41.8781136,
                "longitude": -87.6297982,
                "population": "2718782",
                "rank": "3",
                "state": "Illinois"
            }
        ];
        const searchList = jest.fn();
        const selectCity = jest.fn();

        it('should render multiple <City /> components in <List>', () => {
            const wrapper = mount(<List cities={cities} searchList={searchList} sekectCity={selectCity} />);
            expect(wrapper.find(City)).to.have.lengthOf(3);
        });
    });
    // describe('Map tests', () => {
    //     it('should render map component', () => {
    //         const wrapper = mount(<Map />);

    //         expect(wrapper)
    //     });
    // });
});
