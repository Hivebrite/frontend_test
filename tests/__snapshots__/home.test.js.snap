// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should have 2 components 1`] = `
<div
  className="sc-gKsewC klXKBY"
>
  <div
    className="sc-gsTCUz LBUjF"
  >
    <input
      className="sc-eCssSg kmpSAM"
      onChange={[Function]}
      placeholder="Filter the list"
      type="text"
      value=""
    />
    <div
      className="sc-hKgILt lciAJE"
    >
      <div
        className="sc-dlfnbm fcLTFp"
      />
    </div>
  </div>
  <div
    className="sc-jSgupP dOZTJJ"
  >
    <div />
  </div>
</div>
`;
